### DISCUSSION TITLE
aardvark-dns using port 53

### DISCUSSION THREAD

**Discussion by d-513** [containers/podman#14242] [Q&A] [answered]
Cannot start Pi-Hole when aardvark-dns is enabled, because it takes up port 53
```
[root@archserver containers]# netstat -plntu | grep 53
udp        0      0 10.89.0.1:53            0.0.0.0:*                           2617/aardvark-dns
```
when aardvark-dns is disabled, everything works, but you can't use dns to communicate between containers

Port is hardcoded https://github.com/containers/netavark/blob/4af309fb888d54238dc9028bcfe150425c687b02/src/dns/aardvark.rs#L93-L100

related https://github.com/containers/aardvark-dns/issues/13
anyway to bypass this limitation?

**Comment by mheon** [comment, marked as answer]
Not at present. There is no way to specify that DNS should use a
nonstandard port, so until we implement a NAT solution port 53 must be
used. You can bind your DNS server to specific IPs (the machine’s public
IPs) to avoid the ports bound by Aardvark.

**Reply by d-513** [reply]
> Not at present. There is no way to specify that DNS should use a nonstandard port, so until we implement a NAT solution port 53 must be used. You can bind your DNS server to specific IPs (the machine’s public IPs) to avoid the ports bound by Aardvark.

Thank you! This worked like a charm. For reference for anyone wondering, I published the port to the IP directly: `"192.168.10.225:53:53/udp"`

**Reply by nekename** [reply]
Helped me as well, thanks!

**Comment by tysegall** [comment]
I know this issue is a bit old, but I had to remove the aardvark-dns package from Ubuntu in order to spin up pihole and unbound running on podman with a docker-compse.yml file.  It's a bit of a special case, @mheon but I would say that it "just works" if I'm using legit docker versus podman.   Would be nice if the port was configurable for Aardvark instead of being hardcoded.

**Reply by Luap99** [reply]
The port can be configured in containers.conf
```
[network]
dns_bind_port=5000 # or well whatever you like
```

**Reply by tysegall** [reply]
Thanks!  That actually worked.  I created /etc/containers/containers.conf with a custom port, reinstalled the aardvark-dns package, and I can now successfully start pihole and unbound as defined in the docker-compose.yaml file.  Thank you @Luap99
